waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -0,0 +1,483 @@
+;;
+;;       Copyright (c) 2017 Two Sigma Investments, LP.
+;;       All Rights Reserved
+;;
+;;       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF
+;;       Two Sigma Investments, LP.
+;;
+;;       The copyright notice above does not evidence any
+;;       actual or intended publication of such source code.
+;;
+(ns waiter.scheduler.kubernetes
+  (:require [clj-time.core :as t]
+            [clojure.core.async :as async]
+            [clojure.data.json :as json]
+            [clojure.edn :as edn]
+            [clojure.java.shell :as shell]
+            [clojure.string :as string]
+            [clojure.tools.logging :as log]
+            [plumbing.core :as pc]
+            [slingshot.slingshot :as ss]
+            [waiter.mesos.utils :as mesos-utils]
+            [waiter.scheduler :as scheduler]
+            [waiter.service-description :as sd]
+            [waiter.util.utils :as utils])
+  (:import (org.joda.time.format DateTimeFormat)))
+
+(def service-id->failed-instances-transient-store (atom {}))
```

Comment:
There's no reason this can't be a field in the record. I declared it this way based off the fact that the `service-id->killed-instances-transient-store` in the `waiter.scheduler` module is declared the same way, and declaring this (and the other atoms below) at the module level saves me a bunch of piping. This is one of the points I was expecting to discuss during the first round of feedback.